Support for building on Ubuntu 22.04

Update:

Now that ikos supports llvm 14, there are build issues with python recorded in https://github.com/NASA-SW-VnV/ikos/issues/193#issuecomment-1636683997

---

ikos currently requires llvm 9 which is no longer available in Ubuntu 22.04.

LLVM [11](https://packages.ubuntu.com/jammy/llvm-11), [12](https://packages.ubuntu.com/jammy/llvm-12), [13](https://packages.ubuntu.com/jammy/llvm-13), and [14](https://packages.ubuntu.com/jammy/llvm-14) are all packaged in 22.04. As a side note, Debian Bullseye (the current stable release) provides [LLVM 11](https://packages.debian.org/bullseye/llvm-11) and [13](https://packages.debian.org/bullseye/llvm-13) but not 12 and 14 was not yet released.

It looks like the pending work on LLVM 10 has stalled https://github.com/NASA-SW-VnV/ikos/pull/162.
Looking at the above release coverage, 13 would probably the next version to target in order to build on current (and likely at least one future) releases of Debian and Ubuntu.

Thanks for the message, I gave building another go and hit the following error:

```
/src/ikos/b/analyzer/python/setup.py:42: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.core import setup
Traceback (most recent call last):
  File "/src/ikos/b/analyzer/python/setup.py", line 42, in <module>
    from distutils.core import setup
ModuleNotFoundError: No module named 'distutils.core'
make[2]: *** [analyzer/CMakeFiles/ikos-python.dir/build.make:90: analyzer/python/build/lib/ikos/__init__.py] Error 1
make[1]: *** [CMakeFiles/Makefile2:1644: analyzer/CMakeFiles/ikos-python.dir/all] Error 2
```

If no one is looking at this yet, I think I can at least start a patch to switch away from distutils.

--- a/analyzer/python/setup.py.in
+++ b/analyzer/python/setup.py.in
@@ -39,7 +39,7 @@
 # UNILATERAL TERMINATION OF THIS AGREEMENT.
 #
 ################################################################################
-from distutils.core import setup
+from setuptools import setup
 
 setup(name='ikos',
       version='@PACKAGE_VERSION@',

